guides/source/asset_pipeline.md

```diff
@@ -8,476 +8,613 @@ This guide covers the asset pipeline.
+without manual intervention. This is done as part of the usual Rails asset
+pipeline, with the default configuration ensuring the most recent asset changes
+are served on page reload.
 
-When this option is true, digests will be generated for asset URLs.
+For modern JavaScript bundlers like [esbuild](https://esbuild.github.io/) or
+[Webpack](https://webpack.js.org/) (if used alongside Propshaft), you may run
+the `./bin/dev` command, which watches for changes in JavaScript and CSS files
+and compiles them into the appropriate build directory. Propshaft will detect
+changes in the assets and serve the compiled files to the browser.
 
-### Turning Source Maps On
+#### Improving Performance with File Watchers
 
-You can turn on source maps by updating `config/environments/development.rb` to
-include:
+In development, Propshaft checks if any assets have been updated before each
+request, using the application's file watcher (by default,
+`ActiveSupport::FileUpdateChecker`). If you have a large number of assets, you
+can improve performance by using the `listen` gem and configuring the following
+setting in `config/environments/development.rb`:
 
 ```ruby
-config.assets.debug = true
+config.file_watcher = ActiveSupport::EventedFileUpdateChecker
 ```
 
-When debug mode is on, Sprockets will generate a Source Map for each asset. This
-allows you to debug each file individually in your browser's developer tools.
+This will reduce the overhead of checking for file updates and improve
+performance during development.
 
-Assets are compiled and cached on the first request after the server is started.
-Sprockets sets a `must-revalidate` Cache-Control HTTP header to reduce request
-overhead on subsequent requests - on these the browser gets a 304 (Not Modified)
-response.
+### Production
 
-If any of the files in the manifest change between requests, the server
-responds with a new compiled file.
+In production, Rails serves assets with caching enabled to optimize performance,
+ensuring that your application can handle high traffic efficiently.
 
-In Production
--------------
+#### Asset Caching and Versioning in Production
 
-In the production environment Sprockets uses the fingerprinting scheme outlined
-above. By default Rails assumes assets have been precompiled and will be
-served as static assets by your web server.
+Rails compiles assets into a single location and uses the asset fingerprinting
+method to ensure that users receive the most up-to-date version of the asset.
+You can precompile assets using the `rails:assets:precompile` task. You can read
+ore about this is the [Fingerprinting section](#fingerprinting). By default,
+Rails assumes assets have been precompiled and will be served as static assets
+by your web server.
 
 During the precompilation phase a SHA256 is generated from the contents of the
 compiled files, and inserted into the filenames as they are written to disk.
```

Comment:
I think this could be made a bit more obvious:

```suggestion
compiled files, which inserts a unique string into the filenames as they are written to disk.
```